app/src/main/java/com/github/se/eduverse/repository/NotificationRepository.kt

```diff
@@ -0,0 +1,115 @@
+package com.github.se.eduverse.repository
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import com.github.se.eduverse.model.NotifAutorizations
+import com.github.se.eduverse.model.Scheduled
+import com.github.se.eduverse.model.ScheduledType
+import java.util.concurrent.TimeUnit
+
+// Parameter authorizations is not used for now, it is here to make the class easier to upgrade
+open class NotificationRepository(
+    context: Context,
+    val autorizations: NotifAutorizations,
+    val workManager: WorkManager = WorkManager.getInstance(context)
+) {
+  /**
+   * Prepare a notification for an event or a task
+   *
+   * @param scheduled the event/task scheduled
+   */
+  open fun scheduleNotification(scheduled: Scheduled) {
+    cancelNotification(scheduled)
+
+    val delay = scheduled.start.timeInMillis - System.currentTimeMillis()
+
+    if (delay > 0) {
```

Comment:
Do you mean like 1 minute before, or more like 30 minutes ? Or both ?

Reply:
I think the best would be 4h before since events may take place at remote locations and the user may need some time to get there, so it would be great to be reminded some time in advance so that he doesn't miss it

Reply:
I think it will do